dist/pythonlibs/testrunner/pytest.py

```diff
@@ -20,17 +20,37 @@
 import subprocess
 import re
 import time
+import traceback
 
 import pexpect
 import pytest
 import testrunner
 
-
 DEFAULT_TIMEOUT = 10
 TEST_LOG_CONSOLE = bool(int(os.environ.get('TEST_LOG_CONSOLE', '1')))
 TEST_TIMEOUT_ATTR = 'TIMEOUT'
 
 
+class CustomSpawn(pexpect.spawn):
+    """Convenient subclass to better catch pexpect timeout and eof errors."""
+
+    def expect(self, pattern, timeout=-1, searchwindowsize=-1, async_=False,
+               **kw):
+        # pylint:disable=arguments-differ
+        try:
+            super(CustomSpawn, self).expect(pattern, timeout=timeout,
+                                            searchwindowsize=searchwindowsize,
+                                            async=async_, **kw)
```

Comment:
`async_` was added in `expect 4.3` so this may need to check if this works or if a different name should be used depending on the version https://pexpect.readthedocs.io/en/stable/history.html#version-4-3